src/main/java/org/iac2/service/checking/plugin/implementation/subgraphmatching/comparison/ComponentComparatorForMatchingWithInstanceModel.java

```diff
@@ -0,0 +1,103 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.expression.ExpressionException;
+
+@Getter
+public class ComponentComparatorForMatchingWithInstanceModel implements  SemanticComponentComparator{
+    // todo rethink this if rules are to be modelled based on component names
+    private static List<String> PROPERTIES_TO_IGNORE = new ArrayList<>();
+
+    static {
+        PROPERTIES_TO_IGNORE.add("name");
+    }
+    private static final Logger LOGGER = LoggerFactory.getLogger(ComponentComparatorForMatchingWithInstanceModel.class);
+    private final ComplianceRule rule;
+
+    public ComponentComparatorForMatchingWithInstanceModel(ComplianceRule rule) {
+        this.rule = rule;
+    }
+
+    public ComponentComparisonOutcome compare(RootComponent instanceModelComponent, RootComponent ruleComponent) {
+        if (!ruleComponent.getClass().isAssignableFrom(instanceModelComponent.getClass())) {
+            return ComponentComparisonOutcome.WRONG_TYPE;
+        }
+
+        String expression;
+        String name;
+        boolean isFound;
+        Collection<Property> instanceModelComponentComputedProps = instanceModelComponent
+                .getProperties()
+                .values()
+                .stream()
+                .filter(Property::isComputed)
+                .toList();
+
+        // iterate over all properties of the compliance rule
+        for (Property property : ruleComponent.getProperties().values()) {
+            // only consider the properties that are assigned values
+            // here we cannot use "isComputed" since it is somehow always false when we rebuild the graph from a YAML string!
+            expression = property.getValue();
+            name = property.getName();
+
+            if (expression != null && !expression.isEmpty() && !PROPERTIES_TO_IGNORE.contains(name)) {
+                isFound = false;
+                // iterate over all properties that are assigned values in the instance model
+                for (Property instanceProperty : instanceModelComponentComputedProps) {
+                    // find a property with a suitable name!
+                    if (areEqualPropertyNames(name, instanceProperty.getName())) {
+                        isFound = true;
+
+                        try {
+                            if (!AttributeComparator.evaluateAttribute(expression, instanceProperty, rule)) {
+                                // the attribute values do not match!
+                                return ComponentComparisonOutcome.WRONG_VALUE;
+                            }
+                        } catch (ExpressionException e) {
+                            // expression does not have a boolean value! (we should have discovered this problem earlier!)
+                            LOGGER.error("expression '{}' in property '{}' cannot be evaluated to a boolean value.", expression, name);
+                            return ComponentComparisonOutcome.NOT_BOOLEAN;
```

Comment:
the check here is rather a sanity check.. because we are supposed to check for non-boolean expressions before we even start this method. So the problem is internal (our problem not the users's).